main.go

```diff
@@ -1,9 +1,482 @@
+			case 2:
+				z, _ := strconv.ParseFloat(y, 64)
+				zhel = append(zhel, z)
+			case 4:
+				z, _ := strconv.ParseFloat(y, 64)
+				mb = append(mb, z)
+			case 5:
+				z, _ := strconv.ParseFloat(y, 64)
+				dmb = append(dmb, z)
+			case 6:
+				z, _ := strconv.ParseFloat(y, 64)
+				stretch = append(stretch, z)
+			case 7:
+				z, _ := strconv.ParseFloat(y, 64)
+				dstretch = append(dstretch, z)
+			case 8:
+				z, _ := strconv.ParseFloat(y, 64)
+				colour = append(colour, z)
+			case 9:
+				z, _ := strconv.ParseFloat(y, 64)
+				dcolour = append(dcolour, z)
+			case 10:
+				z, _ := strconv.ParseFloat(y, 64)
+				m_stell = append(m_stell, z)
+			case 17:
+				z, _ := strconv.ParseFloat(y, 64)
+				exp = append(exp, z)
+			case 18:
+				z, _ := strconv.ParseFloat(y, 64)
+				ra_jla = append(ra_jla, z)
+			case 19:
+				z, _ := strconv.ParseFloat(y, 64)
+				de_jla = append(de_jla, z)
+			}
+		}
+	}
+
+	fmt.Println("Number of supernovae : ", len(sn_names))
+
+	// compute and store slices needed for the plots and analysis
+
+	var muexp_data []float64		// experimental mus from SALT2 model
+	muexp_data = mu_exp_slice(m_stell, mb, stretch, colour)
+
+	var muth_data []float64			// theoretical mus from lambdaCDM model
+	muth_data = mu_th_slice(zcmb, 10000)
+
+	var diff_data []float64			// difference between experimental and theoretical mus
+	var abs_diff_data []float64		// absolute values of the differences
+	diff_data, abs_diff_data = difference(muth_data, muexp_data)
+
+	var mean_residuals float64		// mean of the differences
+	mean_residuals = stat.Mean(diff_data, nil)
+
+	var abs_mean_residuals float64		// mean of the absolute differences
+	abs_mean_residuals = stat.Mean(abs_diff_data, nil)
+
+	var muexp_error []float64		// slice containing the error on mu_exp, from error propagation in SALT2 model
+	muexp_error = mu_exp_err(dmb, dstretch, dcolour)
+
+	fmt.Println("mean of the residuals ", mean_residuals)
+	fmt.Println("mean of the absolute residuals ", abs_mean_residuals)
+
+	// fitting function, without covariance matrix
+	/*
+	ps := make([]float64, 0)
+	ps = append(ps, 0.295)
+	res,_ := fit.Curve1D(
+		fit.Func1D{
+			F: func(z float64, ps []float64) float64 {
+				return 5*math.Log10( ((1+z)*299792458/(10*0.070)) * modified_integral(z, ps[0]) )
+			},
+			X: zcmb,
+			Y: muexp_data,
+			Err: muexp_error,
+			Ps: ps,
+		},
+		nil, &optimize.NelderMead{},
+	)
+
+	fmt.Println("Omega M : ", res.X[0])
+*/
+	// compute, draw and store the various diagrams
+
+	scatter_data := points(zcmb, muexp_data)
+	diagram := diag(scatter_data)
+	diagram.Save(1080, 720, "hubble_diagram.png")
+
+	scatter2_data := points(zcmb, diff_data)
+	diagram2 := diag(scatter2_data)
+	diagram2.Save(720, 200, "hubble_diagram2.png")
+
+	histo := histogram(diff_data)
+	histo.Save(500, 500, "histogram.png")
+
+	empty := make([]float64, len(zcmb))
+	params := make([]float64, 0)
+	params = append(params, 0.295, 0.141, 3.101, -19.05, -0.070)
+	res,_ := fit.Curve1D(
```

Comment:
please check for a non=`nil` error returned by `Curve1D`.
you do want to be sure that what you print out as a fit result isn't some garbage (it may very well be if an error occured).